From 7908e2e6895bc2ca3b218cb0562a643a5eb42da7 Mon Sep 17 00:00:00 2001
From: Jooyung Han <jooyung@google.com>
Date: Wed, 16 Oct 2019 22:42:56 +0000
Subject: [PATCH 01/25] Revert "adapt ld.config.txt for vndk apex"

This reverts commit 99457ff3c23797865aecaa4743ca2981e7afe5ff.

Reason for revert: some targets are broken

Bug: 142773030
Change-Id: I5bccf2658d2b7963d2260235da8ec76976475f30
(cherry picked from commit e23a29999f9ff31f7ff2340b3cdf32e563890bc4)
---
 apex/ld.config.txt | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/apex/ld.config.txt b/apex/ld.config.txt
index 5db59e53d1..8f60bb21c9 100644
--- a/apex/ld.config.txt
+++ b/apex/ld.config.txt
@@ -95,7 +95,7 @@ namespace.sphal.asan.permitted.paths +=           /vendor/${LIB}
 
 namespace.sphal.search.paths += /odm/${LIB}/vndk-sp
 namespace.sphal.search.paths += /vendor/${LIB}/vndk-sp
-namespace.sphal.search.paths += /apex/com.android.vndk.v${VNDK_VER}/${LIB}
+namespace.sphal.search.paths += /system/${LIB}/vndk-sp${VNDK_VER}
 
 namespace.sphal.permitted.paths += /odm/${LIB}/hw
 namespace.sphal.permitted.paths += /odm/${LIB}/egl
@@ -104,13 +104,14 @@ namespace.sphal.permitted.paths += /vendor/${LIB}/egl
 namespace.sphal.permitted.paths += /system/vendor/${LIB}/hw
 namespace.sphal.permitted.paths += /system/vendor/${LIB}/egl
 # This is exceptionally required since android.hidl.memory@1.0-impl.so is here
-namespace.sphal.permitted.paths += /apex/com.android.vndk.v${VNDK_VER}/${LIB}/hw
+namespace.sphal.permitted.paths += /system/${LIB}/vndk-sp${VNDK_VER}/hw
 
 namespace.sphal.asan.search.paths += /data/asan/odm/${LIB}/vndk-sp
 namespace.sphal.asan.search.paths +=           /odm/${LIB}/vndk-sp
 namespace.sphal.asan.search.paths += /data/asan/vendor/${LIB}/vndk-sp
 namespace.sphal.asan.search.paths +=           /vendor/${LIB}/vndk-sp
-namespace.sphal.asan.search.paths += /apex/com.android.vndk.v${VNDK_VER}/${LIB}
+namespace.sphal.asan.search.paths += /data/asan/system/${LIB}/vndk-sp${VNDK_VER}
+namespace.sphal.asan.search.paths +=           /system/${LIB}/vndk-sp${VNDK_VER}
 
 namespace.sphal.asan.permitted.paths += /data/asan/odm/${LIB}/hw
 namespace.sphal.asan.permitted.paths +=           /odm/${LIB}/hw
@@ -121,7 +122,8 @@ namespace.sphal.asan.permitted.paths +=           /vendor/${LIB}/hw
 namespace.sphal.asan.permitted.paths += /data/asan/vendor/${LIB}/egl
 namespace.sphal.asan.permitted.paths +=           /vendor/${LIB}/egl
 
-namespace.sphal.asan.permitted.paths += /apex/com.android.vndk.v${VNDK_VER}/${LIB}/hw
+namespace.sphal.asan.permitted.paths += /data/asan/system/${LIB}/vndk-sp${VNDK_VER}/hw
+namespace.sphal.asan.permitted.paths +=           /system/${LIB}/vndk-sp${VNDK_VER}/hw
 
 # Once in this namespace, access to libraries in /system/lib is restricted. Only
 # libs listed here can be used.
-- 
2.36.0

